### PR TITLE
Enhance policy: apply policy's workload before workflow and add context.components for policydefinition

### PR DESCRIPTION
**Pull request by yangsoon**
**What this PR does / why we need it**:

This PR enhance the policy's ability:

1. apply policy's workload before the workflow starts.
2. add `context.components` for policydefinition ,   represents the `components` field in the Application

Others:
1. add `op.#ApplyEnvBindComponent`
2. fix bug of workflow

TODO:

- [x] add test

### CODE REVIEW

File: pkg/cue/process/handle.go [hunk 1]
```diff
@@ -134,7 +139,18 @@ func NewContextWithHooks(namespace, name, appName, appRevision string, baseHooks
 	}
 }
 
-// SetBase set templateContext base model
+// NewPolicyContext create Application Scope templateContext for Policy
+func NewPolicyContext(namespace, name, appName, appRevision string, components []common.ApplicationComponent) Context {
+	return &templateContext{
+		name:        name,
```
Comment (wonderflow) [inline pkg/cue/process/handle.go:145]: name is the component name, which should not be in context of Policy

Comment (yangsoon) [reply pkg/cue/process/handle.go:145]: this name represent the name of policy

Comment (wonderflow) [reply pkg/cue/process/handle.go:145]: ok, that's fine

**Review by wonderflow — APPROVED** [review]
GOOD JOB

**Review by wonderflow — APPROVED** [review]

**Pull request merged**